### ISSUE TITLE
Courses section of Front Page never finishes loading

### ISSUE THREAD

**Issue by michael-milette** [dualcube/moodle-theme_roshnilite#9]
When viewing the Front page, the _Courses_ section of the page shows a spinning graphic as if it was loading something but it never ends and nothing ever appears.

![image](https://cloud.githubusercontent.com/assets/3808579/11004294/651b7f46-8485-11e5-8651-6bd70dd412ce.png)

Whether I am logged in or logged out, I get the same result.

I am using Moodle 2.7.10+ with the November 6th version of the theme downloaded from Moodle.org.

**Comment by parthajeet** [comment]
Is there any folder permission issue in your server? We tried to replicate the same in our environment but it's working correctly. It will be helpful if you share your server and moodle access details with us.

**Comment by michael-milette** [comment]
There are no folder permission issues that I am aware of... unless you are trying to create temporary files anywhere other than the moodledata directory. We also do not have web services enabled in case that is relevant.

Like I said previously, it works fine with 30+ other themes. Just not with yours.

I realize that it would be helpful to share our server and Moodle access with you but that is not possible.

Best regards,

```
Michael
```

**Comment by parthajeet** [comment]
Hi,

We have tried for a considerable number of environments and with the last version of our theme, we couldn't find this issue in any one if them. Hence, we are not considering it as bug unless any other also point this same issue. i hope you will understand.

**Comment by pitervergara** [comment]
;tldr
I'm facing the same issue and I think it's related with broken images on this section. So, how can I identify/define which images are used to represent the course in this section of the front page?

## 

I'm facing the same issue in a 3.0.3 installation. 
What I could notice so far is that this section is creating img tags which the "src" points to files that are not available on my site and, therefore, some HTTP404 are showing up. Perhaps not been able to deal with these 404 is what creates the behavior. 
Follows an example of HTML rendered on my frontpage:

```
<ul class="av-courses-slider">
    <li class = "view fourth-effect">
        <img src="http://ead.astroceap.com.br/pluginfile.php/50/course/legacy/ZeusN.GIF" alt="<< THIS FILE DOES NOT EXISTS >>">
        <div class="mask"></div>
        <div class="av-course-item-cont">
            <h2><a href="http://ead.astroceap.com.br/course/view.php?id=5">CURSO II -  ZODÍACO</a></h2>
        </div>
    </li>
    ...
</ul>    
```

The course URL is correct and if go there I don't see any references to that ''ZeusN.GIF'' file in the main course page.

Important to note that this image really does no exist among this course legacy files, (which I know it's a problem beyond the scope of the theme) and even if I upload this file, it still cannot be reached via the URL used in the image src.

Also, the files seems to be randomly chosen, because I have some src attributes pointing to ".html" and ".pdf" files. 

Please, can someone point out how can I identify/define which images are used to represent the course in the front page? Perheaps if I'm abble to define a diferent image, this wilI solve the issue.

**Comment by parthajeet** [comment]
Hello Piter,

Please only upload picture in course summary files in jpg/png format as in front end there is image slider. So it cannot render html/pdf files.

Below is the site map for course summary files:- 

Home / ► {your category} / ► {your course} / ► Edit settings > description > Course summary files 

Regards,
Parthajeet~

**Comment by yatesg** [comment]
I am having the same issues as Michael-Millette at the top of the section. Course section empty, spinning graphic.  Moodle 3.03. No permission or ownership issues anywhere. SELinux set to permissive.

**Comment by pitervergara** [comment]
Thanks, Parthajeet
it works. After solving the http404 the courses are correctly displayed

**Comment by parthajeet** [comment]
Hi Piter,

Glad to know that this took care of your problem.

Regards,
Parthajeet~

**Comment by parthajeet** [comment]
Hello Garry,

Can you please provide us a screenshot of the homepage after enabling the console in browser. To enable console you need to press f12 and then go to the console tab(in firefox / chrome).

Thanks,
Parthajeet~

**Comment by yatesg** [comment]
Parthajeet,

I've attached a screen shot with the console errors showing.  The spinning wheel on the homppage with the Roshnilite theme

Latest firefox, Linux Fedora 23 OS, Moodle 3.03

Thanks

Garry

**Comment by parthajeet** [comment]
Hello Garry,

The attachment is missing. Can you please check once.

Thanks,
Parthajeet~

**Comment by yatesg** [comment]
I've attached the screenshot again.  Called Roshnilite_moodle3_R10.png

Hopefully it works this time

Thanks

**Comment by LAAO** [comment]
Hello Parthajeet,

I had the same issue as michael-milette because the picture was uploaded in course summary files in jpg format, but when I deleted that picture, the courses were shown in the slider. 
Nevertheless, I have now a different issue, because the courses can't have any pictures (even if I upload them in Course summary). Any idea about how to solve this issue?
My Moodle version is 3.2.1, and Roshni Lite Release 2.5 (Build: 2015092114).

Many thanks!



![courses](https://user-images.githubusercontent.com/29398943/27078513-2c08c624-5034-11e7-9345-1d1f130b5fbe.JPG)

**Comment by kndnsow** [comment]
it's solved and added in the latest version 3.4
